overlord/hookstate/hookmgr.go

```diff
@@ -143,13 +153,60 @@ func (m *HookManager) doRunHook(task *state.Task, tomb *tomb.Tomb) error {
 		return err
 	}
 
-	// TODO: Actually dispatch the hook.
+	// Actually run the hook
+	_, err = RunHook(setup.Snap, setup.Revision, setup.Hook, tomb)
+	if err != nil {
+		if handlerErr := handler.Error(err); handlerErr != nil {
+			return handlerErr
+		}
 
-	// Done with the hook. TODO: Check the result, if success call Done(), if
-	// error, call Error(). Since we have no hooks, for now we just call Done().
-	if err := handler.Done(); err != nil {
+		return err
+	}
+
+	// Assuming no error occurred, notify the handler that the hook has
+	// finished.
+	if err = handler.Done(); err != nil {
 		return err
 	}
 
 	return nil
 }
+
+func doRunHook(snapName string, revision snap.Revision, hookName string, tomb *tomb.Tomb) ([]byte, error) {
```

Comment:
Can you re-expose it in the follow up please, so we have some more context and can review why it's necessary there?